src/huntsman/pocs/camera/zwo.py

```diff
@@ -177,6 +183,27 @@ def reconnect(self):
         self._reset_usb()
         return self.connect()
 
+    def take_exposure(self, defocused=False, *args, **kwargs):
+        """ Overrride class method to add defocusing offset. """
+        if defocused:
+            required_focus_offset = self._defocus_offset - self._current_defocus_offset
+            self.logger.debug(f"Applying focus offset of {self._defocus_offset} for defocused "
```

Comment:
I'd move this line down to where you actually move the focus.

Also the logs will be filled with a bunch of "Nullifying defocus offset" as `defocused=False`.

You could just detect one of 2 cases on line 197: `required_focus_offset` is positive or negative and you print either "Nullifying defocus offset" or "Applying focus offset of {self._defocus_offset} for defocused " depending on whether it is postive or negative?

Reply:
The required focus offset depends on whether the exposure is defocused or not, hence it is encapsulated within the `if` statement. The `nullifying focus offset` will appear in the debug log for every exposure - this is the sort of thing we want in the debug logs.